### PR TITLE
docs(LH-70404): Address documentation errors identified by Bernice

### PR DESCRIPTION
**Pull request by siddhuwarrier**
https://jira-eng-rtp3.cisco.com/jira/browse/LH-70404

### Description

Address documentation errors and nits identified by docs team.

### CODE REVIEW

**Review by hazital — APPROVED** [review]
LGTM

File: provider/internal/user/resource.go [hunk 1]
```diff
@@ -62,18 +62,18 @@ func (r *Resource) Metadata(ctx context.Context, req resource.MetadataRequest, r
...
 				Computed:            true,
 				PlanModifiers: []planmodifier.String{
 					stringplanmodifier.UseStateForUnknown(),
 				},
 			},
 			"name": schema.StringAttribute{
-				MarkdownDescription: "The username. If the user is not an [API only user](https://www.cisco.com/c/en/us/td/docs/security/cdo/managing-ftd-with-cdo/managing-ftd-with-cisco-defense-orchestrator/basics-of-cisco-defense-orchestrator.html?bookSearch=true#Cisco_Task.dita_d5ae397b-5aa5-4de0-82c1-a4aff63c5ba1), it must be an e-mail address. Otherwise, it must not be an e-mail address.",
+				MarkdownDescription: "The username. If the user is not an [API only user](https://www.cisco.com/c/en/us/td/docs/security/cdo/managing-ftd-with-cdo/managing-ftd-with-cisco-defense-orchestrator/basics-of-cisco-defense-orchestrator.html?bookSearch=true#Cisco_Task.dita_d5ae397b-5aa5-4de0-82c1-a4aff63c5ba1), it must be an e-mail address; if the user is an API-only user, it must not be an email address, and CDO will generate a name for the user prefixed by the value provided here (see `generated_username`).",
```
Comment (weilueluo) [inline provider/internal/user/resource.go:76]: is `generated_username` supposed to be a link?

**Review by weilueluo — APPROVED** [review]
LGTM! Only one comment

File: provider/internal/user/resource.go (hunk 1 above)
Comment (siddhuwarrier) [reply provider/internal/user/resource.go:76]: No, this is a Terraform module specific thing, and the link is to the input below.

**Pull request merged**